relayer/src/keyring.rs

```diff
@@ -380,19 +400,31 @@ fn private_key_from_mnemonic(
 }
 
 /// Return an address from a Public Key
-fn get_address(pk: ExtendedPubKey) -> Vec<u8> {
-    let mut hasher = Sha256::new();
-    hasher.update(pk.public_key.to_bytes().as_slice());
+fn get_address(pk: ExtendedPubKey, at: &AddressType) -> Vec<u8> {
+    match at {
+        AddressType::Cosmos => {
+            let mut hasher = Sha256::new();
+            hasher.update(pk.public_key.to_bytes().as_slice());
+
+            // Read hash digest over the public key bytes & consume hasher
+            let pk_hash = hasher.finalize();
+
+            // Plug the hash result into the next crypto hash function.
+            let mut rip_hasher = Ripemd160::new();
+            rip_hasher.update(pk_hash);
+            let rip_result = rip_hasher.finalize();
 
-    // Read hash digest over the public key bytes & consume hasher
-    let pk_hash = hasher.finalize();
+            rip_result.to_vec()
+        }
+        AddressType::Ethermint { .. } => {
+            let public_key = pk.public_key.key.serialize_uncompressed();
+            debug_assert_eq!(public_key[0], 0x04);
```

Comment:
Just some minor things - would be nice to document that the `0x04` corresponds to [SECP256K1_TAG_PUBKEY_UNCOMPRESSED](https://github.com/bitcoin-core/secp256k1/blob/d7ec49a6893751f068275cc8ddf4993ef7f31756/include/secp256k1.h#L196) and that we're only taking the right-most 20 bytes of the hash for the address.